File: src/main/java/club/xiaojiawei/custom/MouseClickListener.java
package club.xiaojiawei.custom;

import java.awt.event.MouseEvent;
import java.awt.event.MouseListener;
import java.util.function.Consumer;

/**
 * @author 肖嘉威
 * @date 2023/10/1 10:12
 * @question <a href=""/>
 */
public class MouseClickListener implements MouseListener {

    private Consumer<MouseEvent> consumer;

    public MouseClickListener(Consumer<MouseEvent> consumer) {
        this.consumer = consumer;
    }

    @Override
    public void mouseClicked(MouseEvent e) {
        consumer.accept(e);
    }

    @Override
    public void mousePressed(MouseEvent e) {

    }

    @Override
    public void mouseReleased(MouseEvent e) {

    }

    @Override
    public void mouseEntered(MouseEvent e) {

    }

    @Override
    public void mouseExited(MouseEvent e) {

    }
}

File: hs-script-app/src/main/java/club/xiaojiawei/hsscript/custom/MouseClickListener.kt
package club.xiaojiawei.hsscript.custom

import java.awt.event.MouseEvent
import java.awt.event.MouseListener
import java.util.function.Consumer

/**
 * @author 肖嘉威
 * @date 2023/10/1 10:12
 */
class MouseClickListener(private val consumer: Consumer<MouseEvent?>) : MouseListener {

    override fun mouseClicked(e: MouseEvent?) {
        consumer.accept(e)
    }

    override fun mousePressed(e: MouseEvent?) {
    }

    override fun mouseReleased(e: MouseEvent?) {
    }

    override fun mouseEntered(e: MouseEvent?) {
    }

    override fun mouseExited(e: MouseEvent?) {
    }
}
